app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt

```diff
@@ -330,6 +334,13 @@ class SearchFragment : Fragment(), UserInteractionHandler {
             )
         }
 
+        // Users can from this fragment go to install/uninstall search engines and then return.
+        with(areShortcutsAvailable()) {
```

Comment:
I don't think `with` is the right tool here, since there are no properties to work with on a boolean. We're better off just setting a `val` like above.
```suggestion
        val areShortcutsAvailable = areShortcutsAvailable()
        if (searchStore.state.areShortcutsAvailable != areShortcutsAvailable) {
            searchStore.dispatch(SearchFragmentAction.UpdateShortcutsAvailability(areShortcutsAvailable))
        }
```

Reply:
Thanks!
Maybe a personal preference of mine to use `with()` instead of declaring a val.
Indeed, not in the spirit of what `with()` should be used for. Will change.